browser/extensions/brave_extension_provider.cc

```diff
@@ -9,98 +9,26 @@
 #include <vector>
 
 #include "base/strings/utf_string_conversions.h"
+#include "brave/browser/brave_browser_process_impl.h"
 #include "brave/common/extensions/extension_constants.h"
+#include "brave/components/brave_shields/browser/extension_whitelist_service.h"
 #include "brave/grit/brave_generated_resources.h"
 #include "ui/base/l10n/l10n_util.h"
 
 namespace {
 
 bool IsBlacklisted(const extensions::Extension* extension) {
-  // This is a hardcoded list of extensions to block.
-  // Typically instead you can just use the brave/go-updater to list
-  // a blacklisted extension that you want to block for existing clients.
-  static std::vector<std::string> blacklisted_extensions({
-    // Used for tests, corresponds to brave/test/data/should-be-blocked-extension.
-    "mlklomjnahgiddgfdgjhibinlfibfffc",
-    // Chromium PDF Viewer.
-    "mhjfbmdgcfjbbpaeojofohoefgiehjai"
```

Comment:
I think since `mhjfbmdgcfjbbpaeojofohoefgiehjai` is a built in component we can't load it remotely because of a race condition on startup.  Note that we still want the ability for blacklist as you have done it, so even know I'm suggesting we keep both of these hardcoded exceptions, we still want the ability to add things fast if we need to in the blacklist.